sbi/utils/diagnostics_utils.py

```diff
@@ -91,3 +93,20 @@ def sample_fun(
         posterior_samples.shape[:2]
     }."""
     return posterior_samples
+
+
+def remove_nans_and_infs(thetas: Tensor, xs: Tensor) -> Tuple[Tensor, Tensor]:
```

Comment:
Might also be worth to check thetas for infs or nans. This is probably less common, but the function name suggests that all nans and infs are removed (otherwise maybe rename to remove_nan_and_inf_xs)